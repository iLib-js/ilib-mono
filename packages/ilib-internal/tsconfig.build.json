{
    "extends": "./config/tsconfig.build.json",
    "includes": ["src"]
}
