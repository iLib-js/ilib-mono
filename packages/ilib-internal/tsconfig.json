{
    "extends": "./config/tsconfig.base.json",
    "includes": ["src"]
}
