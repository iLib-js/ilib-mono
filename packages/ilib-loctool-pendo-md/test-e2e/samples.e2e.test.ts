/*
 * samples.e2e.test.js - E2E tests for ilib-loctool-po samples
 *
 * Copyright Â© 2025, Box, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import path from "path";
import fs from "fs";
import { expectFileToMatchSnapshot, LoctoolRunner, FSSnapshot } from "ilib-internal";

describe("samples", () => {
    describe("pendo-md", () => {
        let fsSnapshot: FSSnapshot;
        const projectPath = path.resolve(__dirname, "..", "samples", "pendo");
        const pathInProject = (p: string) => path.resolve(projectPath, p);

        const projectFiles = {
            loctool: {
                extracted: pathInProject("ilib-loctool-pendo-md-test-extracted.xliff"),
                new: pathInProject("ilib-loctool-pendo-md-test-new.xliff"),
            },
            pendo: {
                source: pathInProject("l10n/xliff/guides/A000A00Aaa0aaa-AaaaAaa00A0a_en-US.xliff"),
                localized: pathInProject("l10n/xliff/guides/A000A00Aaa0aaa-AaaaAaa00A0a_en-US_pl-PL.xliff"),
            },
        };

        beforeAll(async () => {
            const filesToSnapshot = Object.values(projectFiles).flatMap((f) => Object.values(f));
            fsSnapshot = FSSnapshot.create(filesToSnapshot);

            // run loctool
            const loctool = new LoctoolRunner(projectPath);
            await loctool.run("localize");
        });

        afterAll(() => {
            fsSnapshot.restore();
        });

        it("should produce an extracted Loctool XLIFF file", () => {
            expectFileToMatchSnapshot(projectFiles.loctool.extracted);
        });

        it("should not produce a newly-extracted Loctool XLIFF file", () => {
            expect(fs.existsSync(projectFiles.loctool.new)).toBe(false);
        });

        it("should produce a localized Pendo XLIFF file", () => {
            expectFileToMatchSnapshot(projectFiles.pendo.localized);
        });
    });
});
