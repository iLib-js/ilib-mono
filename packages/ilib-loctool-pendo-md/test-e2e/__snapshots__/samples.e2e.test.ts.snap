// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`samples pendo-md should produce a localized Pendo XLIFF file 1`] = `
"<xliff version="1.2" xmlns="urn:oasis:names:tc:xliff:document:1.2">
    <file original="A000A00Aaa0aaa-AaaaAaa00A0a" datatype="pendoguide" source-language="en-US" target-language="pl-PL">
        <body>
            <group id="Aaaaaaaa0aAaaAAA0AAA0A0aAaa">
                <note/>
                <trans-unit id="73e6c197-f1c0-4bb1-9b31-a40ce2270870|ariaLabel">
                    <source><![CDATA[Close]]></source>
                    <target state="translated">Zamknij</target>
                    <note>CloseButton</note>
                </trans-unit>
                <trans-unit id="8de49842-c1fd-4536-905e-8817673b4c24|md">
                    <source><![CDATA[**Callout!**]]></source>
                    <target state="translated">**Wywołanie!**</target>
                    <note>TextView</note>
                </trans-unit>
                <trans-unit id="a165c0e2-11de-40b4-9209-55bcdff6d67b|md">
                    <source><![CDATA[First paragraph of a multi-paragraph string.

Second paragraph of a multi-paragraph string.]]></source>
                    <target state="translated">Pierwszy akapit wieloakapitowego ciągu.

Drugi akapit wieloakapitowego ciągu.
        </target>
                    <note>TextView</note>
                </trans-unit>
                <trans-unit id="5f8bec10-d663-43d7-a185-49824564060c|md">
                    <source><![CDATA[String with [a link in it](https://example.com)]]></source>
                    <target state="translated">Ciąg z [linkiem wewnątrz](https://example.com)</target>
                    <note>TextView</note>
                </trans-unit>
                <trans-unit id="cff9a58d-39e6-4562-8b39-96a137d0bfd1|url">
                    <source><![CDATA[https://example.com/linkOnly]]></source>
                    <target state="translated">https://example.com/linkOnly</target>
                    <note>ButtonLink</note>
                </trans-unit>
                <trans-unit id="7542337f-62c3-44c3-9a4a-d0f7944dd5b4|md">
                    <source><![CDATA[First paragraph of a multi-paragraph string with list.

This is a list:
* first item
* second item
* third item]]></source>
                    <target state="translated">Pierwszy akapit wieloakapitowego ciągu z listą.

To jest lista:

*   pierwszy element

*   drugi element

*   trzeci element</target>
                    <note>TextView</note>
                </trans-unit>
            </group>
        </body>
    </file>
</xliff>"
`;

exports[`samples pendo-md should produce an extracted Loctool XLIFF file 1`] = `
"<?xml version="1.0" encoding="utf-8"?>
<xliff version="1.2">
  <file original="" source-language="en-US" product-name="pendo-md">
    <body>
      <trans-unit id="1" resname="73e6c197-f1c0-4bb1-9b31-a40ce2270870|ariaLabel" restype="string" datatype="plaintext">
        <source>Close</source>
        <note>CloseButton</note>
      </trans-unit>
      <trans-unit id="2" resname="8de49842-c1fd-4536-905e-8817673b4c24|md" restype="string" datatype="plaintext">
        <source>&lt;c0&gt;Callout!&lt;/c0&gt;</source>
        <note>TextView [c0: strong]</note>
      </trans-unit>
      <trans-unit id="3" resname="a165c0e2-11de-40b4-9209-55bcdff6d67b|md" restype="string" datatype="plaintext">
        <source>First paragraph of a multi-paragraph string.

Second paragraph of a multi-paragraph string.</source>
        <note>TextView</note>
      </trans-unit>
      <trans-unit id="4" resname="5f8bec10-d663-43d7-a185-49824564060c|md" restype="string" datatype="plaintext">
        <source>String with &lt;c0&gt;a link in it&lt;/c0&gt;</source>
        <note>TextView [c0: link]</note>
      </trans-unit>
      <trans-unit id="5" resname="cff9a58d-39e6-4562-8b39-96a137d0bfd1|url" restype="string" datatype="plaintext">
        <source>https://example.com/linkOnly</source>
        <note>ButtonLink</note>
      </trans-unit>
      <trans-unit id="6" resname="7542337f-62c3-44c3-9a4a-d0f7944dd5b4|md" restype="string" datatype="plaintext">
        <source>First paragraph of a multi-paragraph string with list.

This is a list:

&lt;c0&gt;

&lt;c1&gt;

first item

&lt;/c1&gt;

&lt;c2&gt;

second item

&lt;/c2&gt;

&lt;c3&gt;

third item

&lt;/c3&gt;

&lt;/c0&gt;</source>
        <note>TextView [c0: list, c1: listItem, c2: listItem, c3: listItem]</note>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;
